fx-src/FirefoxStorage.js

```diff
@@ -72,7 +72,12 @@ const statements = {
   "listRecords": `
     SELECT record
       FROM collection_data
-        WHERE collection_name = :collection_name;`
+        WHERE collection_name = :collection_name;`,
+
+  "importData": `
+    REPLACE INTO collection_data (collection_name, record_id, record)
```

Comment:
… is at odds with this. You iterate over the incoming records in a transaction, calling `INSERT OR REPLACE` with each. That will unconditionally discard the existing rows that violate the uniqueness constraint on `record_id`:

```
When a UNIQUE or PRIMARY KEY constraint violation occurs, the REPLACE algorithm deletes pre-existing rows that are causing the constraint violation prior to inserting or updating the current row and the command continues executing normally.
```